Kernel/Locking/Spinlock.h

```diff
@@ -70,7 +70,7 @@ class [[nodiscard]] SpinlockLocker {
 
 private:
     LockType* m_lock { nullptr };
-    u32 m_prev_flags { 0 };
+    InterruptsState m_interrupts_state_before_locking { InterruptsState::Disabled };
```

Comment:
nitpick: This seems extremely verbose for what it is, as I mentioned elsewhere, I think the "previous" wording made sense, and could be extended for clarity, without naming it something so verbose?

Reply:
I named it `m_previous_interrupts_state` for now :)